### PR TITLE
[SPARK-6658][SQL] Update DataFrame documentation to fix type references.

### PR DESCRIPTION
**Pull request by chetmancini**
First contribution here; would love to be getting some code contributions in soon. Let me know if there's anything about contribution process I should improve.

### CODE REVIEW

**Comment by marmbrus** [comment]
ok to test

**Comment by rxin** [comment]
Actually this looks great. Thanks @chetmancini.  Will merge it after the tests pass.

**Comment by marmbrus** [comment]
```
[error] /home/jenkins/workspace/SparkPullRequestBuilder@2/sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala:979: File line length exceeds 100 characters
```

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala [hunk 1]
```diff
@@ -976,7 +976,7 @@ class DataFrame private[sql](
   def javaRDD: JavaRDD[Row] = toJavaRDD
 
   /**
-   * Registers this RDD as a temporary table using the given name.  The lifetime of this temporary
+   * Registers this [[DataFrame]] as a temporary table using the given name.  The lifetime of this temporary
```
Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala]: This line is now over 100 characters. We'd need to wrap it.

**Comment by chetmancini** [comment]
retest this please

**Comment by marmbrus** [comment]
Thanks!  I've merged this to master and branch-1.3

**Pull request closed without merging**